lib/assets/assets.go

```diff
@@ -53,14 +55,18 @@ func Serve(w http.ResponseWriter, r *http.Request, asset Asset) {
 		return
 	}
 
-	if strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
+	header.Set("Content-Length", strconv.Itoa(asset.Length))
+
+	switch {
+	case !asset.Gzipped:
+		io.WriteString(w, asset.Content)
+	case strings.Contains(r.Header.Get("Accept-Encoding"), "gzip"):
 		header.Set("Content-Encoding", "gzip")
-		header.Set("Content-Length", strconv.Itoa(len(asset.ContentGz)))
-		io.WriteString(w, asset.ContentGz)
-	} else {
+		io.WriteString(w, asset.Content)
```

Comment:
Content-length gone?

Reply:
It's above

Reply:
Wait. We're now sending the length of the uncompressed asset with the compressed data? does that work?

Reply:
No. Sorry, I was mistaken. It turns out [Content-Length is the encoded length](https://stackoverflow.com/a/3819303).